File: eps/src/main/java/com/company/leaveshanghai/LeaveShanghaiService.java
package com.company.leaveshanghai;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class LeaveShanghaiService {

    private final LeaveShanghaiRepository leaveShanghaiRepository;

    @Autowired
    public LeaveShanghaiService(LeaveShanghaiRepository leaveShanghaiRepository) {
        this.leaveShanghaiRepository = leaveShanghaiRepository;
    }

    public List<LeaveShanghai> getLeaveShanghaiApplications() { return leaveShanghaiRepository.findAll(); }

    public List<LeaveShanghai> getLeaveShanghaiApplicationsByStudentId(Integer studentId) {
        return leaveShanghaiRepository.findAllByStudentId(studentId);
    }

    public void addNewLeaveShanghaiApplication(LeaveShanghai leaveShanghai) {
        var newLeaveShanghaiApplication = LeaveShanghai.builder()
                .username(leaveShanghai.getUsername())
                .studentId(leaveShanghai.getStudentId())
                .email(leaveShanghai.getEmail())
                .department(leaveShanghai.getDepartment())
                .leaveDate(leaveShanghai.getLeaveDate())
                .enterDate(leaveShanghai.getEnterDate())
                .province(leaveShanghai.getProvince())
                .city(leaveShanghai.getCity())
                .destinationAddress(leaveShanghai.getDestinationAddress())
                .applicationReason(leaveShanghai.getApplicationReason())
                .travelTrajectory(leaveShanghai.getTravelTrajectory())
                .transportation(leaveShanghai.getTransportation())
                .applicationType(leaveShanghai.getApplicationType())
                .submitDate(leaveShanghai.getSubmitDate())
                .status(0)
                .build();

        leaveShanghaiRepository.save(newLeaveShanghaiApplication);
    }

    public void changeStatus(Integer studentId, LocalDateTime submitDate, Integer status) {
        List<LeaveShanghai> leaveShanghaiList = leaveShanghaiRepository.findAllByStudentId(studentId);
        for (LeaveShanghai leaveShanghai : leaveShanghaiList) {
            if (leaveShanghai.getSubmitDate().toString().equals(submitDate.toString())) {
                leaveShanghai.setStatus(status);
                leaveShanghaiRepository.save(leaveShanghai);
                break;
            }
        }
    }
}

File: eps/src/frontend/src/services/leave-shanghai.service.js
import axios from 'axios';

const API_URL = '/api/v1/leave-shanghai';

class LeaveShanghaiService {
    submit(leaveShanghai) {
        return axios
            .post(API_URL + '/submit', {
                username: leaveShanghai.username,
                studentId: leaveShanghai.studentId,
                email: leaveShanghai.email,
                department: leaveShanghai.department,
                applicationType: leaveShanghai.applicationType,

                leaveDate: leaveShanghai.leaveDate,
                enterDate: leaveShanghai.enterDate,
                province: leaveShanghai.province,
                city: leaveShanghai.city,
                destinationAddress: leaveShanghai.destinationAddress,
                applicationReason: leaveShanghai.applicationReason,
                travelTrajectory: leaveShanghai.travelTrajectory,
                transportation: leaveShanghai.transportation,
                submitDate: leaveShanghai.submitDate,
                status: 0
            })
    }
    getAll() {
        return axios
            .get(API_URL)
            .then(response => {
                return response;
            })
    }
    getAllByStudentId(studentId) {
        return axios
            .get(API_URL + '/' + studentId, {
                studentId: studentId
            })
            .then(response => {
                return response;
            })
    }
    changeStatus(leaveShanghai, status) {
        return axios
            .put(API_URL + "/" +leaveShanghai.studentId + "?submitDate=" + leaveShanghai.submitDate.replace(" ", "T") + "&status=" + status)
            .then(response => {
                return response;
            })
    }
}

export default new LeaveShanghaiService();
